src/execution/mod.rs

```diff
@@ -21,24 +21,37 @@ mod stats;
 use std::io::Write;
 use std::path::PathBuf;
 use std::sync::Arc;
+use std::time::Duration;
 
 use log::{error, info};
 pub use stats::{collect_plan_stats, ExecutionStats};
 
 use crate::config::ExecutionConfig;
 use crate::extensions::{enabled_extensions, DftSessionStateBuilder};
-use color_eyre::eyre::Result;
+use color_eyre::eyre::{self, Result};
 use datafusion::execution::SendableRecordBatchStream;
-use datafusion::physical_plan::ExecutionPlan;
+use datafusion::physical_plan::{execute_stream, ExecutionPlan};
 use datafusion::prelude::*;
-use datafusion::sql::parser::Statement;
+use datafusion::sql::parser::{DFParser, Statement};
 use tokio_stream::StreamExt;
 #[cfg(feature = "flightsql")]
 use {
     crate::config::FlightSQLConfig, arrow_flight::sql::client::FlightSqlServiceClient,
     tokio::sync::Mutex, tonic::transport::Channel,
 };
 
+// Container for benchmark statistics for a single location execution
+//
+// Note: It would be nice if we could split out planning and optimization for both logical and
+// physical planning but its not obvious from the DataFusion APIs how to do that.
```

Comment:
It would be nice if you could do something like the below because then we could time how long each step takes.

```rust
let logical_plan = self.statement_to_plan(statement);
let optimized_logical_plan = self.optimize_logical_plan(logical_plan);
let physical_plan = self.create_physical_plan(optimized_logical_plan);
let optimized_physical_plan = self.optimize_physical_plan(physical_plan);

// Continue with executing the optimized physical plan
```

Reply:
@alamb do you have any thoughts on the above?

Reply:
It is possible to do this -- most of the underlying apis are on `SessionState` I believe

Reply:
I saw `create_logical_plan` and `optimize` to go from sql to optimized logical plan, and then i saw `create_physical_plan` which itself optimizes the logical plan but i didnt see a way to go from logical plan to physical plan without optimize being automatically called on the logical plan and then i didnt see way to explicitly call optimize on the physical plan.